app/src/utils/btcsnap-signer.ts

```diff
@@ -295,3 +297,23 @@ async function transferInscription(
 
   return broadcastTx(electrsClient, tx.txHex);
 }
+
+/**
+ * Given an array of utxos passed in, return those that do not contain any inscriptions.
+ */
+async function findUtxosWithoutInscriptions(
+  ordinalsClient: OrdinalsClient,
+  utxos: UTXO[]
+): Promise<UTXO[]> {
+  const safeUtxos = [];
+
+  for (const utxo of utxos) {
+    // can be optimized later
+    const inscriptionUtxo = await ordinalsClient.getInscriptionFromUTXO(`${utxo.txid}:${utxo.vout}`);
```

Comment:
Ordinals won't be able to check mempool transactions but we can optimize later

Reply:
> Ordinals won't be able to check mempool transactions but we can optimize later

I had a suspicion that might be the case.
That's why I added the `confirmed: true` flag when getting all utxos. I think that should limit the query set to those that ordinals would have sufficient information about.

Reply:
Ah I see that now, okay I have the change to allow us to check also unconfirmed utxos will add that in a separate PR